### PR TITLE
Finished working on task 6, GET /api/articles/:article_id/comments - …

### PR DESCRIPTION
**Pull request by jckdvnprt**
GET /api/articles/:article_id/comments
made sure that they're returned in order of newest first
added endpoints to json file and added tests for various outcomes

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -121,46 +137,99 @@ describe("GET /api/articles/:article_id", () => {
...
-          );
-          for (let i = 1; i < retrievedArticles.length; i++) {
-            expect(
-              new Date(retrievedArticles[i].created_at).getTime()
-            ).toBeLessThanOrEqual(
-              new Date(retrievedArticles[i - 1].created_at).getTime()
-            );
-          }
+          expect(200);
+          expect(Array.isArray(response.body)).toBe(true);
```
Comment (sirKevlar) [inline __tests__/app.test.js:147]: Here it is vital to check that the array length is greater than 0 or to assert against the actual expected array length

File: __tests__/app.test.js [hunk 2]
```diff
@@ -121,46 +137,99 @@ describe("GET /api/articles/:article_id", () => {
...
         .then((response) => {
-          const retrievedArticles = response.body;
-          retrievedArticles.forEach((article) => {
-            expect(article).toHaveProperty("comment_count");
-          });
+          const retrievedComments = response.body;
+
+          expect(200);
+          for (let i = 1; i < retrievedComments.length; i++) {
+            expect(
```
Comment (sirKevlar) [inline __tests__/app.test.js:169]: Jest sorted would do this job and is what we would recommend using here

File: __tests__/app.test.js [hunk 3]
```diff
@@ -121,46 +137,99 @@ describe("GET /api/articles/:article_id", () => {
...
         });
+      test("Get comments for non-existent article id", () => {
+        const nonExistentArticleId = 13;
+        return request(app)
+          .get(`/api/articles/${nonExistentArticleId}/comments`)
+          .then((response) => {
+            expect(response.status).toBe(404);
+            expect(response.body).toEqual({ error: "Article not found" });
+          });
+      });
```
Comment (sirKevlar) [inline __tests__/app.test.js:184]: We need a test for an invalid user ID eg /articles/not-a-valid-id/comments

File: __tests__/app.test.js [hunk 4]
```diff
@@ -121,46 +137,99 @@ describe("GET /api/articles/:article_id", () => {
...
+          const retrievedComments = response.body;
+
+          expect(200);
+          for (let i = 1; i < retrievedComments.length; i++) {
+            expect(
+              retrievedComments[i].created_at.toBeLessThanOrEqual(
+                retrievedComments[i - 1].created_at
+              )
+            );
+          }
```
Comment (sirKevlar) [inline __tests__/app.test.js:174]: One extra happy path test here for a valid article with no associated comments status 200 returns an empty array

File: db/controllers/controller.js [hunk 5]
```diff
@@ -11,6 +12,14 @@ const getTopics = (req, res, next) => {
   });
 };
 
+const getComments = (req, res, next) => {
+  const articleId = req.params.article_id; // Correctly access article_id from req.params
```
Comment (sirKevlar) [inline db/controllers/controller.js:16]: Remove comments please

File: db/models/model.js [hunk 6]
```diff
@@ -53,8 +53,25 @@ function fetchArticleFromDatabase(articleId) {
   });
 }
 
+function fetchCommentsFromDatabase(articleId) {
+  return new Promise((resolve, reject) => {
```
Comment (sirKevlar) [inline db/models/model.js:57]: PSQL is already returning a resolved or rejected promise here no need to create a new instance of the promise class
